tests/Benchmark/EntryPoint.cpp

```diff
@@ -0,0 +1,157 @@
+            netRuntime += results[i];
+            mean += (results[i] - mean) / (i + 1);
+            q += (results[i] - previousMean) * (results[i] - mean);
+        }
+        Microseconds sampleStandardDeviation = std::sqrt(q / (runCount - 1));
+        m_results.emplace_back(__Result{ caseName, netRuntime, mean, sampleStandardDeviation, runCount });
+    }
+
+protected:
+    std::vector<__Result> m_results;
+};
+
+class CSVBenchmarkPublisher : public PublisherBase {
+    std::string m_outPath;
+
+public:
+    CSVBenchmarkPublisher(std::string&& outPath) : m_outPath(outPath) {
+    }
+
+    virtual void PublishResults() {
+        std::ofstream outFile(m_outPath);
+        outFile << "Case Name"
+                << ","
+                << "Total Runtime"
+                << ","
+                << "Mean Runtime"
+                << ","
+                << "Standard Deviation"
+                << ","
+                << "Run Count"
+                << "\n";
+        for (auto res : m_results) {
+            outFile << res.testName << "," << res.totalRuntime << "," << res.meanRuntime << "," << res.stdDeviation << "," << res.runCount
+                    << "\n";
+        }
+
+        outFile.flush();
+        outFile.close();
+        m_results.clear();
+    }
+};
+
+class LogBenchmarkPublisher : public PublisherBase {
+    size_t m_maxNameWidth = 0;
+
+public:
+    virtual void RegisterCaseResults(const std::string& testName, const std::vector<Microseconds>& results) {
+        m_maxNameWidth = max(m_maxNameWidth, testName.size());
+        PublisherBase::RegisterCaseResults(testName, results);
+    }
+
+    virtual void PublishResults() {
+        // Align output horizontally
+        std::string spaces(m_maxNameWidth - 5, ' ');
+        std::cout << "Case Name" << spaces << "|"
+                  << "Total Runtime"
+                  << "\t\t|"
+                  << "Mean Runtime"
+                  << "\t\t|"
+                  << "Standard Deviation"
+                  << "\t|"
+                  << "Run Count"
+                  << "\n";
+        for (auto res : m_results) {
+            std::string spaces(m_maxNameWidth - res.testName.size() + 4, ' ');
+            std::cout << std::fixed << res.testName << spaces << "|" << res.totalRuntime << "\t\t|" << res.meanRuntime << "\t\t|"
+                      << res.stdDeviation << "\t\t|" << res.runCount << "\n";
+        }
+
+        std::cout << std::endl;
+        m_results.clear();
+    }
+};
+
+static std::shared_ptr<BenchmarkPublisher> s_publisher;
+void BenchmarkPublisherFactory::CreatePublisher(int argc, char** argv) {
+    // Currently only support CSV format
+    bool created = false;
+    for (int i = 1; i < argc && argv[i]; ++i) {
+        char* arg = argv[i];
+        if (strncmp(arg, "--out=", 6) == 0) {
+            s_publisher.reset(new CSVBenchmarkPublisher(std::move(std::string(arg + 6))));
+            created = true;
+        }
+    }
+
+    if (!created) {
```

Comment:
(instead of keeping track of this silly thing.)

---
In reply to: [99419315](https://github.com/Microsoft/WinObjC/pull/1876#discussion_r99419315) [](ancestors = 99419315)